src/api/java/baritone/api/utils/command/Command.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * This file is part of Baritone.
+ *
+ * Baritone is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Baritone is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Baritone.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package baritone.api.utils.command;
+
+import baritone.api.BaritoneAPI;
+import baritone.api.IBaritone;
+import baritone.api.Settings;
+import baritone.api.event.listener.AbstractGameEventListener;
+import baritone.api.utils.Helper;
+import baritone.api.utils.IPlayerContext;
+import baritone.api.utils.command.execution.CommandExecution;
+import baritone.api.utils.command.helpers.arguments.ArgConsumer;
+import net.minecraft.client.Minecraft;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public abstract class Command implements Helper, AbstractGameEventListener {
+    protected IBaritone baritone = BaritoneAPI.getProvider().getPrimaryBaritone();
+    protected Settings settings = BaritoneAPI.getSettings();
+    protected IPlayerContext ctx = baritone.getPlayerContext();
+    protected Minecraft MC = mc;
```

Comment:
Just don't.

Reply:
In other words, the most that should be done here is have a `baritone` and `ctx` field. As well as rather than having the `baritone` field be the primary instance strictly, the command should take in the associated instance somehow.

Reply:
Yes, taking in the Baritone instance would fix the above Travis issue with the BaritoneProvider.

Reply:
aids